fw-tools/edge-testnet

```diff
@@ -327,44 +338,88 @@ main() {
     fi
 }
 
-displayHelp() {
+usage() {
     echo "Usage: $0 -options"
     echo "  -d do not delete temporary storage"
     echo "  -v verbose output"
+    echo "  -e debug"    
+    echo "  -s skip cert validation in TLS tests"
+    echo "  -h/-H show this help dialog"
+    echo "  --domain  domain name"
+    echo "  --env environment. Can be integration/os2/production"
+    echo "  etiher --domain or --env must be supplied"
     exit
 }
 
 argprocessor() {
-    while getopts "hHdev" optsin; do
-        case "${optsin}" in
+    args=$(getopt -o hHdevs --l udp,env:,domain: -- "$@")
+    
+    if [[ -z "$args" ]]; then
+        usage
+    fi
+
+    eval set -- "${args}"
+    
+    while :
+    do
+        case $1 in
+            -d) DONTDELETE=1; shift ;;
+            #
+            -e) DEBUG=1; shift ;;
             #
-            d) DONTDELETE=1 ;;
+            -h) displayHelp; shift ;;
             #
-            e) DEBUG=1 ;;
+            -H) displayHelp; shift ;;
             #
-            h) displayHelp ;;
+            -v) VERBOSE=1; shift ;;
             #
-            H) displayHelp ;;
+            -s) SKIP_CERT_VALID=1; shift;;
             #
-            v) VERBOSE=1 ;;
+            --udp) TEST_UDP=1; shift;;
             #
-            \?)
-                echo -e "Option -$OPTARG not allowed.\n "
-                displayHelp
-                ;;
-                #
+            --env) ENV="$2"; shift 2 ;;
+            #
+            --domain) DOMAIN_NAME="$2"; shift 2 ;;
+            #
+            --) shift; break ;;
+            *) >&2 echo Unsupported option: "$1"
+              usage ;;
         esac
     done
-    shift $((OPTIND - 1))
-    if [[ $# -ne 0 ]]; then
-        displayHelp
-    else
-        shift
-        # set echo on if DEBUG=1
-        if [[ "$DEBUG" -eq 1 ]]; then
-            set -x
+    
+    # set echo on if DEBUG=1
+    if [[ "$DEBUG" -eq 1 ]]; then
+        set -x
+    fi
+    
+    # Check if either --env or --domain arguments were provided
+    if [[ -z "$ENV" && -z "$DOMAIN_NAME" ]]; then
+        echo "Both --env or --domain argument weren't supplied."
```

Comment:
I think if no env is given, that we test against production.
To stay backwards compatible.

Reply:
Fixed